From d03d3719d96f2c2a36d53e08e32e38aa830d8672 Mon Sep 17 00:00:00 2001
From: "Fabio M. Di Nitto" <fdinitto@redhat.com>
Date: Tue, 24 Apr 2012 13:51:36 +0200
Subject: [PATCH] rgmanager: cleanup binary linking and include dirs

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>
Reviewed-by: Lon Hohberger <lhh@redhat.com>
Reviewed-by: Ryan McCabe <rmccabe@redhat.com>
---
 rgmanager/src/daemons/Makefile |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/rgmanager/src/daemons/Makefile b/rgmanager/src/daemons/Makefile
index 336b6cf..d125eaa 100644
--- a/rgmanager/src/daemons/Makefile
+++ b/rgmanager/src/daemons/Makefile
@@ -49,7 +49,8 @@ OBJS5=	cpglockdump.o
 
 CFLAGS += -DSHAREDIR=\"${sharedir}\" -D_GNU_SOURCE
 CFLAGS += -fPIC
-CFLAGS += -I${ccsincdir} -I${cmanincdir} -I${dlmincdir} -I${logtincdir}
+CFLAGS += -I${ccsincdir} -I${cmanincdir} -I${corosyncincdir}
+CFLAGS += -I${dlmincdir} -I${fencedincdir} -I${logtincdir}
 CFLAGS += `xml2-config --cflags` -I${slangincdir}
 CFLAGS += -I$(S)/../../include
 CFLAGS += -I${incdir}
@@ -63,8 +64,8 @@ CCS_LDFLAGS += -L${ccslibdir} -lccs
 CMAN_LDFLAGS += -L${cmanlibdir} -lcman
 LOGSYS_LDFLAGS += -L${logtlibdir} -llogthread
 DLM_LDFLAGS += -L${dlmlibdir} -ldlm
-CPG_LDFLAGS += -lcpg
-FENCED_LDFLAGS += -lfenced
+CPG_LDFLAGS += -L${corosynclibdir} -lcpg
+FENCED_LDFLAGS += -L${fencedlibdir} -lfenced
 XML2_LDFLAGS += `xml2-config --libs`
 SLANG_LDFLAGS += -L${slanglibdir} -lslang
 EXTRA_LDFLAGS += -lpthread
-- 
1.7.7.6

