From 79b02f3654e438a216cdb86ccda735b83563fe00 Mon Sep 17 00:00:00 2001
From: Ryan McCabe <rmccabe@redhat.com>
Date: Mon, 23 Apr 2012 18:27:55 -0400
Subject: [PATCH] Update Makefile for cpglockd init script

Signed-off-by: Ryan McCabe <rmccabe@redhat.com>
Reviewed-by: Lon Hohberger <lhh@redhat.com>
Reviewed-by: Fabio M. Di Nitto <fdinitto@redhat.com>
---
 rgmanager/init.d/Makefile |   19 +++++++++++++------
 1 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/rgmanager/init.d/Makefile b/rgmanager/init.d/Makefile
index 4c9a44f..7346884 100644
--- a/rgmanager/init.d/Makefile
+++ b/rgmanager/init.d/Makefile
@@ -1,19 +1,26 @@
-TARGET= rgmanager
+TARGET1=rgmanager
+TARGET2=cpglockd
 
-INITDT=$(TARGET)
+INITDT=$(TARGET1) $(TARGET2)
 
-all: $(TARGET)
+all: $(TARGET1) $(TARGET2)
 
 include ../../make/defines.mk
 include $(OBJDIR)/make/clean.mk
 include $(OBJDIR)/make/install.mk
 include $(OBJDIR)/make/uninstall.mk
 
-$(TARGET): $(S)/$(TARGET).in
-	cat $(S)/$(TARGET).in | sed \
+$(TARGET1): $(S)/$(TARGET1).in
+	cat $(S)/$(TARGET1).in | sed \
 		-e 's#@SBINDIR@#${sbindir}#g' \
 		-e 's#@INITDDIR@#${initddir}#g' \
-	> $(TARGET)
+	> $(TARGET1)
+
+$(TARGET2): $(S)/$(TARGET2).in
+	cat $(S)/$(TARGET2).in | sed \
+		-e 's#@SBINDIR@#${sbindir}#g' \
+		-e 's#@INITDDIR@#${initddir}#g' \
+	> $(TARGET2)
 
 clean: generalclean
 
-- 
1.7.7.6

